src/rate/index.tsx

```diff
@@ -0,0 +1,343 @@
+import { DNode, RenderResult } from '@dojo/framework/core/interfaces';
+import { uuid } from '@dojo/framework/core/util';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import focus from '@dojo/framework/core/middleware/focus';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import Icon from '../icon';
+import Radio from '../radio';
+import RadioGroup from '../radio-group';
+import theme from '../middleware/theme';
+import * as baseCss from '../common/styles/base.m.css';
+import * as iconCss from '../theme/default/icon.m.css';
+import * as radioCss from '../theme/default/radio.m.css';
+import * as radioGroupCss from '../theme/default/radio-group.m.css';
+import * as css from '../theme/default/rate.m.css';
+import bundle from './nls/Rate';
+
+export interface MixedNumber {
+	quotient: number;
+	numerator: number;
+	denominator: number;
+}
+
+export interface RateProperties {
+	/** The name attribute for this rating group */
+	name: string;
+	/** The label to be displayed in the legend */
+	label?: string;
+	/** The id used for the radio input elements */
+	widgetId?: string;
+	/** The initial rating value */
+	initialValue?: number;
+	/** Callback fired when the rating value changes */
+	onValue?: (value?: number, mixedValue?: MixedNumber) => void;
```

Comment:
is this `mixedValue` really required? It does not line up with other `onValue` functions that pass only a single value.

Reply:
It's not strictly required, but it makes turning decimal values into fractions a little difficult without it.

Reply:
value is now returned within the object